src/main/java/address/util/UrlUtil.java

```diff
@@ -16,7 +17,36 @@ public static boolean compareBaseUrls(URL url1, URL url2) {
         boolean isSameBaseUrl = url1.getHost().toLowerCase().replaceFirst("www.", "")
                                 .equals(url2.getHost().replaceFirst("www.", "").toLowerCase())
                                 && url1.getPath().replaceAll("/", "").toLowerCase()
-                                   .equals(url2.getPath().replaceAll("/", "").toLowerCase());
+                                .equals(url2.getPath().replaceAll("/", "").toLowerCase());
```

Comment:
URL class might have methods that could make this comparison simpler. Try googling for a better solution.